src/control/tools/InputHandler.cpp

```diff
@@ -32,7 +31,7 @@ void InputHandler::resetShapeRecognizer() {
     // Does nothing here. Implemented in the extending classes
 }
 
-void InputHandler::createStroke(Point p) {
+void InputHandler::createStroke(Point p, const std::lock_guard<std::recursive_mutex>&) {
```

Comment:
**General Todo for a refactoring:** I am not happy with the naming: createXXX should either return a Stroke or a unique / shared pointer to a Stroke. Not void.

Reply:
this for example is a perfect way to show, that the locking is too wide. The only thing, which must be protected is the line 37: `stroke = new Stroke();`
also this line can be moved to the end of the function / the return value by replacing the line with `auto stroke = std::make_unique<Stroke>(); ...; this->stroke = stroke.release();`